benchmarks/src/jmh/java/software/amazon/jdbc/benchmarks/ReadWriteSplittingLoadBenchmarks.java

```diff
@@ -0,0 +1,197 @@
+  private static final String PASSWORD = "password";
+
+  private static final int NUM_THREADS = 10;
+  private static final int EXECUTE_QUERY_TIMES = 5;
+  protected static final String QUERY_1 = "select " +
+      "l_returnflag, " +
+      "l_linestatus, " +
+      "sum(l_quantity) as sum_qty, " +
+      "sum(l_extendedprice) as sum_base_price, " +
+      "sum(l_extendedprice * (1 - l_discount)) as sum_disc_price, " +
+      "sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge, " +
+      "avg(l_quantity) as avg_qty, " +
+      "avg(l_extendedprice) as avg_price, " +
+      "avg(l_discount) as avg_disc, " +
+      "count(*) as count_order " +
+      "from " +
+      "LINEITEM " +
+      "where " +
+      "l_shipdate <= date '1998-12-01' - interval '110' day " +
+      "group by " +
+      "l_returnflag, " +
+      "l_linestatus " +
+      "order by " +
+      "l_returnflag, " +
+      "l_linestatus;";
+
+  @Setup(Level.Iteration)
+  public static void setUp() throws SQLException {
+    if (!org.postgresql.Driver.isRegistered()) {
+      org.postgresql.Driver.register();
+    }
+
+    if (!Driver.isRegistered()) {
+      Driver.register();
+    }
+  }
+
+  protected static Properties initNoPluginPropsWithTimeouts() {
+    final Properties props = new Properties();
+    props.setProperty(PGProperty.USER.getName(), USERNAME);
+    props.setProperty(PGProperty.PASSWORD.getName(), PASSWORD);
+    props.setProperty(PGProperty.TCP_KEEP_ALIVE.getName(), Boolean.FALSE.toString());
+    props.setProperty(PGProperty.CONNECT_TIMEOUT.getName(), "5");
+    props.setProperty(PGProperty.SOCKET_TIMEOUT.getName(), "5");
+
+    return props;
+  }
+
+  protected static Properties initReadWritePluginProps() {
+    final Properties props = initNoPluginPropsWithTimeouts();
+    props.setProperty(PropertyDefinition.PLUGINS.name, "auroraHostList,readWriteSplitting");
+    return props;
+  }
+
+  protected static Properties initReadWritePluginLoadBalancingProps() {
+    final Properties props = initNoPluginPropsWithTimeouts();
+    props.setProperty(PropertyDefinition.PLUGINS.name, "auroraHostList,readWriteSplitting");
+    props.setProperty(ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.name, "true");
+    return props;
+  }
+
+  protected static Connection connectToInstance(String instanceUrl, Properties props)
+      throws SQLException {
+    return DriverManager.getConnection(instanceUrl, props);
+  }
+
+  @Benchmark
+  public void noPluginEnabledBenchmarkTest() {
+    final List<Thread> connectionThreadsList = new ArrayList<>(NUM_THREADS);
+
+    for (int i = 0; i < NUM_THREADS; i++) {
+      connectionThreadsList.add(getThread_PGReadWriteSplitting(initNoPluginPropsWithTimeouts()));
+    }
+
+    for (Thread connectionThread : connectionThreadsList) {
+      connectionThread.start();
+    }
+
+    for (Thread connectionThread : connectionThreadsList){
+      connectionThread.interrupt();
```

Comment:
I'm not super familiar with jmh, but why ware we interrupting the threads right after starting them? Won't this stop the thread from completing its task?

Reply:
it stops the thread after it's task is complete

Reply:
I thought that calling interrupt stops the thread wherever its at in its task (complete or not). How do we ensure the task is complete before interrupt is called?